docs/source/textdata.rst

```diff
@@ -165,20 +177,35 @@ It uses the following special chars:
 
           A special case exists for ``\\v`` - if immediately followed by a digit 0 - 9, the underlying base value
           is interpreted as an array of values and the nth element is retrieved where `n` is the digit specified.
-          
+
           The ``escapeSpecialChars`` is set to False by default for backwards compatibility.
 
           To use the ``escapeSpecialChars`` option, use the variant
           ``text=dg.TemplateGenerator(template=...), escapeSpecialChars=True``
 
-In all other cases, the char itself is used.
 
-The setting of the ``escapeSpecialChars`` determines how templates generate data.
+Using a custom word list
+^^^^^^^^^^^^^^^^^^^^^^^^
 
-If set to False, then the template ``r"\\dr_\\v"`` will generate the values ``"dr_0"`` ... ``"dr_999"`` when applied
-to the values zero to 999. This conforms to earlier implementations for backwards compatibility.
+The template generator allows specification of a custom word list also. This is a list of words that can be
+used in the template generation. The default word list is the `ipsum lorem` word list.
 
-If set to True, then the template ``r"dr_\\v"`` will generate the values ``"dr_0"`` ... ``"dr_999"``
-when applied to the values zero to 999. This conforms to the preferred style going forward
+While the `values` option allows for the specification of a list of categorical values, this is transmitted as part of
+the generated SQL. The use of the `TemplateGenerator` object with a custom word list allows for specification of much
+larger lists of possible values without the need to transmit them as part of the generated SQL.
+
+
+Other forms of text value lookup
+--------------------------------
+
+The use of the `values` option and the `template` option with a `TemplateGenerator` instance allow for generation of
+data when the range of possible values is known.
+
+But what about scenarios when the list of data is read from a different table or some other form of lookup?
+
+As the output of the data generation `build()` method is a regular PySpark DataFrame, it is possible to join the
+generated data with other data sources to generate the required data.
 
+In these cases, the generator can be specified to produce lookup keys that can be used to join with the
```

Comment:
Can we get an example of how to do this?

Reply:
lets do this in a later checkin

Reply:
added